docs/Design/Null Checking.md

```diff
@@ -0,0 +1,148 @@
+# Null Checking
+
+Creates checks for if the passed variable has value null. 
+
+A null variable is one that has no other value stored in it.
+
+Can be used to check whether the value is or is not null. 
+
+## Commands
+
+### `is null`
+
+`is null : variable`
+
+Creates a test that will return true if the variable is `null`.
+
+### `not null`
+
+`not null : variable`
+
+Creates a test that will return false if the variable is `null`.
+
+## Usage
+
+```
+is null : foo
+```
+
+```
+not null : bar

```

Comment:
I would actually prefer `is not null` over `not null`. These will likely be implemented as `IsNullCommand` and `NotNullCommand`/`IsNotNullCommand`, so keeping them alphabetically close is good.

Also, `is null` has slightly different grammatical meaning than `not null`. `is not null` is more similar.